bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/items/Metadata.java

```diff
@@ -0,0 +1,87 @@
+/**
+ * Copyright (c) 2014-2017 by the respective copyright holders.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.smarthome.core.items;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.core.common.registry.Identifiable;
+
+/**
+ * This is a data class for storing meta-data for a given item and namespace.
+ * It is the entity used for within the {@link MetadataRegistry}.
+ *
+ * @author Kai Kreuzer - Initial contribution and API
+ *
+ */
+@NonNullByDefault
+public class Metadata implements Identifiable<MetadataKey> {
+
+    private final MetadataKey key;
+    private final String value;
+    private final Map<String, Object> configuration;
+
+    public Metadata(MetadataKey key, String value, @Nullable Map<String, Object> configuration) {
+        this.key = key;
+        this.value = value;
+        this.configuration = configuration != null ? new HashMap<>(configuration) : Collections.emptyMap();
+    }
+
+    @Override
+    public MetadataKey getUID() {
+        return key;
+    }
+
+    /**
+     * Provides the configuration meta-data.
+     *
+     * @return configuration as a map of key-value pairs
+     */
+    public Map<String, Object> getConfiguration() {
+        return Collections.unmodifiableMap(configuration);
```

Comment:
Isn't the configuration of the `Metadata` itself already fixed / unmodifiable?
In the constructor `Collections.emptyMap()` is used if no configuration is provided, so in that case the member is already unmodifiable. We could change the constructor part if a provided configuration is assigned from `new HashMap<>(configuration)` to `Collections.unmodifiableMap(new HashMap<>(configuration))` and return `configuration` here.
As we return a view this already requires that the collection itself is not altered, because this could break (CME) the usage of the returned view.